src/common.rs

```diff
@@ -240,7 +240,7 @@ impl Frame<'static> {
         let mut colors_vec: Vec<(u8, u8, u8, u8)> = colors.into_iter().collect();
         colors_vec.sort();
```

Comment:
This sort doesn't seem necessary, or it could be unstable?

Reply:
I'm not completely sure either. I think it helps a bit with compression of the images by making similar colors have a similar index in the palette. That would actually be interesting to try out because sorting is certainly not an optimal way to achieve that. If you have some days to spend on optimizing this :D

Reply:
Although pure LZW compression should not be affected by the indices anyways, its length is invariant to permutation of symbols afaik.

Reply:
That was my thought process as well, I might look into this a bit but would have to come up with a benchmark.

Nearby, there may be room for improvement at the cost of making the logic more complex. A HashSet is used, collected into a Vec, then a HashMap is built from that Vec. I imagine the hashing is expensive but I don't have any numbers to show. The HashMap and palette indexing could be built from the start but there'd be a mental overhead of handling the indexing bookkeeping.
https://github.com/image-rs/image-gif/blob/033353951d73be44d532720bd00c80617d04da17/src/common.rs#L220-L243

If I don't get around to it, I'll try to file an issue.

Reply:
Open an issue on it right away, that way we at least keep track of this.